### PR TITLE
Add: Avalon CeDeFi of Avalon Finance

### PR DESCRIPTION
**Pull request by tpktpj**
# Avalon CeDeFi 

- Intro: user can deposit **Ignition FBTC**, receive collateral token `lfbtc-Avalon-eth` and borrow USDT.

- TVL Methodology: `lfbtc` collateral and USDT debt of Avalon CeDefi pool contract. 
- `lfbtc` represent the amount of supplied FBTC 

## Protocol Update request

- Add the **Avalon CeDeFi** as a sub-protocol under Avalon Finance(https://defillama.com/protocol/avalon-finance#information)
- Add the Avalon CeDeFi TVL to Avalon Finance
- Verify the `lfbtc-Avalon-eth` token, treat it as **Ignition FBTC** 
----

## What's `lfbtc`? 

- `lfbtc` is  1:1 to Ignition FBTC(https://www.coingecko.com/en/coins/ignition-fbtc). 

- `lfbtc` can be used as collateral in the CeDeFi pool of Avalon Finance. 

- Deposit FBTC to CeDeFi pool, FBTC burnt, lfbtc can minted. 

- Example Supply FBTC tx: https://etherscan.io/tx/0x8ed4ebd29ad695db5c5e085e6631f632edac750c72d7bb3b6c74b64cbaf9f576

- Example Borrow USDT tx: https://etherscan.io/tx/0x6b99b747ab78a4f1aa0a7ea15c032004fe1314821c74fa6263da6186826ab8c0

## Contracts of Avalon CeDeFi

- CeDeFi Lending Pool: 
https://etherscan.io/address/0x02feDCff97942fe28e8936Cdc3D7A480fdD248f0

- lfbtc-Avalon-eth:
https://etherscan.io/address/0x3119a1AD5B63A000aB9CA3F2470611eB997B93B9

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/avalon-finance-cedefi exports TVL: 
        
 
 
    ethereum-borrowed         0
    borrowed                  0
    ethereum                  0
    
    total                    0

**Comment by tpktpj** [comment]
```
Balance table for [ethereum] Unrecognized tokens
┌─────────┬────────────────────┬────────────────────┬─────────┬───────────────────────────────────────────────────────┬──────────┐
│ (index) │ name               │ symbol             │ balance │ label                                                 │ decimals │
├─────────┼────────────────────┼────────────────────┼─────────┼───────────────────────────────────────────────────────┼──────────┤
│ 0       │ 'lfbtc-Avalon-eth' │ 'lfbtc-Avalon-eth' │ '0'     │ 'ethereum:0x3119a1ad5b63a000ab9ca3f2470611eb997b93b9' │ '8'      │
└─────────┴────────────────────┴────────────────────┴─────────┴───────────────────────────────────────────────────────┴──────────┘
```

**Comment by llamatester** [comment]
The adapter at projects/avalon-finance-cedefi exports TVL: 
        
 
 
    ethereum-borrowed         0
    borrowed                  0
    ethereum                  0
    
    total                    0

**Comment by g1nt0ki** [comment]
I am bit lost, when user converts FBTC to LFBTC, is it automatic? why is FBTC burnt instead of locked in a contract?

What happens when the amount of USDT borrowed is higher than the collateral, how are the users liquidated?

**Review by g1nt0ki — CHANGES_REQUESTED** [review]
@tpktpj thanks for the PR, is the protocol active? also check my earlier comment

**Comment by tpktpj** [comment]
HI @g1nt0ki, can you our founder DM @jzpp_1234 on Telegram? He will explain the LFBTC to you, thanks!

**Comment by tpktpj** [comment]
When borrowers pledge FBTC into our pool, we perform the burning of FBTC / minting of LFBTC simultaneously. It will also trigger a transfer of corresponding amount of underlying BTC from a master address to a dedicated address reserved as lending collateral. 

For different use cases (eg. Avalon lending, Babylon staking), a different “Locked FBTC” will be minted, and the specific Locked FBTC will be mapped to underlying BTC in dedicated addresses. 

FBTC can be locked in dedicated contract. But FBTC team design it as “burn standard FBTC and mint equivalent amount of Locked FBTC” to segregate risks among different use cases. 

Pls refer to Section 3.3 of https://fbtc.com/home/FBTC-Whitepaper.pdf

In short, LFBTC represents the amount of FBTC (and also the underlying BTC) pledged in our lending pool.

In the event of liquidation, the lender can trigger liquidation of the respective amount of native BTC to restore the LTV back to healthy level. Once the native BTC is sold/liquidated, the same amount of LFBTC will be forced burned, to maintain the 1:1 peg of LFBTC to underlying BTC.

**Comment by tpktpj** [comment]
Avalon CeDeFi market is already live: https://cedefi.avalonfinance.xyz/

**Comment by tpktpj** [comment]
We would like to add the CeDeFi Market as a sub-protocol under the Avalon Labs. 
```
Avalon Labs: 
- Avalon Finance
- Avalon CeDeFi
```

**Comment by llamatester** [comment]
The adapter at projects/avalon-finance-cedefi exports TVL: 
        
 
 
    bitcoin                   57.21 M
    ethereum-borrowed         28.52 M
    borrowed                  28.52 M
    ethereum                  0
    
    total                    57.21 M

**Comment by tpktpj** [comment]
@g1nt0ki Hi, we have added the Avalon CeDeFi BTC TVL address on the Bitcoin chain.

```
BTC（BTC） : 32DgQPVHSV6FSxLnw68nggvchp3ZNKquxA
BTC（BTC） : bc1qr5nz7n8ulcdz0w3s6fska80fawxhvqlh273qypm3rkjequ9wpmhs65ppw7
BTC（BTC） : bc1qhu98nf6ddz6ja73rn72encdr8ezsyhexwpdzap0vcs7lg2wpmrnq5ygfsl
BTC（BTC） : bc1qg6asmzjr7nr5f5upg3xqyrdxl2tq8ef58hha7t0s82mzzx6zjxesyccp4h
BTC（BTC） : bc1qxe3md4lehg8gmrlx3e8xqju5mytt266l4hcy8khl6tm5mahghmeqtxlgqq
BTC（BTC） : bc1qy48h0kuv0r3e330wjfs6r74sk49pkzumnm907t5mhqjxml22r3ss2ucuxc
```

**Comment by g1nt0ki** [comment]
hi @tpktpj thanks for the update, these bitcoin addresses represent the lfBTC backing?  if so, you can skip counting on ethereum as well?

**Comment by tpktpj** [comment]
@g1nt0ki Hi, YES, the bitcoin addresses represent the lfBTC backing. The code has been updated, we skipped the lfBTC count on Ethereum and counted BTC on bitcoin instead.

**Comment by llamatester** [comment]
The adapter at projects/avalon-finance-cedefi exports TVL: 
        
 
 
    bitcoin                   57.02 M
    ethereum-borrowed         28.53 M
    borrowed                  28.53 M
    ethereum                  0
    
    total                    57.02 M

**Comment by tpktpj** [comment]
@g1nt0ki Hi! Could you please add this adapter (Avalon CeDeFi) as a new product under our current protocol(https://defillama.com/protocol/avalon-finance)?

Also, we've finished our rebranding, the protocol name and logo are updated. 

The protocol name is Avalon Labs. Rebranding PR: https://github.com/DefiLlama/defillama-server/pull/7932#discussion_r1744468502

----

We want to display our protocol 2 product in this format:

Protocol Name: Avalon Labs
- Product1: Avalon Fiannce (https://github.com/DefiLlama/DefiLlama-Adapters/blob/main/projects/avalon-finance/index.js)
- Product2: Avalon CeDeFi (https://github.com/DefiLlama/DefiLlama-Adapters/blob/main/projects/avalon-finance-cedefi/index.js)
